src/js/components/InventoryListItem.js

```diff
@@ -0,0 +1,210 @@
+  }
+
+  removeFromCart() {
+
+    if (Credentials.isProblemUser()) {
+      // Bail out now, don't remove from cart if the item ID is even
+      if (this.state.id % 2 === 0) {
+        return;
+      }
+    }
+
+    ShoppingCart.removeItem(this.state.id);
+    this.setState({ itemInCart: false });
+  }
+
+  navigateToItem() {
+
+    var itemId = this.state.id;
+    if (Credentials.isProblemUser()) {
+      itemId += 1;
+    }
+
+    this.navigation.navigate('InventoryItem', { id: itemId });
+  }
+
+  render() {
+
+    var cartButton;
+
+    if (ShoppingCart.isItemInCart(this.state.id)) {
+      cartButton = (
+        <Button
+          buttonStyle={ [ styles.button_style, styles.remove_button_style ] }
+          containerStyle={ styles.button_container_style }
+          titleStyle={ [ styles.button_title_style, styles.remove_button_title_style ] }
+          onPress={ this.removeFromCart }
+          title={ i18n.t('inventoryItemPage.removeButton') }
+          { ...testProperties(i18n.t('inventoryListPage.removeButton')) }
+        />);
+    } else {
+      cartButton = (
+        <Button
+          buttonStyle={ styles.button_style }
+          containerStyle={ styles.button_container_style }
+          titleStyle={ styles.button_title_style }
+          onPress={ this.addToCart }
+          title={ i18n.t('inventoryItemPage.addButton') }
+          { ...testProperties(i18n.t('inventoryListPage.addButton')) }
+        />);
+    }
+    // Needed to add `accessible={false}` to the TouchableOpacity component, otherwise the components are not accessible and shown
+    // in a flat UI structure
+    return (
+      <View style={ styles.item_container }{ ...testProperties(i18n.t('inventoryListPage.itemContainer')) }>
+        <TouchableOpacity
+          onPress={ this.navigateToItem }
+          { ...MAKE_ACCESSIBLE_FOR_AUTOMATION }
+        >
+          <Image
+            source={ this.state.image_url }
+            style={ styles.item_image }
+            resizeMode="contain"
+          />
+
+          <View { ...testProperties(i18n.t('inventoryListPage.itemDescription')) }>
+            <Text style={ styles.item_name }>{ this.state.name }</Text>
+            <Text style={ styles.item_desc }>{ this.state.desc }</Text>
+          </View>
+
+          <Divider style={ [ styles.divider, styles.description_price_divider ] }/>
+
+          <View>
+            <Text style={ styles.price_text }{ ...testProperties(i18n.t('inventoryListPage.price')) }>
+              ${ this.state.price }
+            </Text>
+            { cartButton }
+          </View>
+
+          <Divider style={ styles.divider }/>
+
+        </TouchableOpacity>
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  item_container: {
+    flexDirection: 'row',
+    paddingLeft: 25,
+    paddingRight: 25,
+    paddingTop: 40,
+  },
+  item_image: {
+    flex: 1,
+    alignSelf: 'stretch',
+    // This is for keeping the aspect ratio and make it repsonse
```

Comment:
typos: "make it repsonse" -> "making it responsive"

Reply:
Fixed